ext/random/random.c

```diff
@@ -558,14 +563,15 @@ PHPAPI int php_random_bytes(void *bytes, size_t size, bool should_throw)
 			) {
 				close(fd);
 				if (should_throw) {
-					zend_throw_exception(zend_ce_exception, "Error reading from source device", 0);
+					zend_throw_exception(zend_ce_exception, "Error reading from /dev/urandom", 0);
```

Comment:
you think you can catch errno here too ? e.g. file descriptor limit.

Reply:
Ah, indeed. `fstat` might also fail. Thanks, adjusted.